lldb/source/Plugins/Language/Swift/SwiftUnsafeTypes.cpp

```diff
@@ -423,9 +423,80 @@ bool lldb_private::formatters::swift::UnsafeTypeSummaryProvider(
   return true;
 }
 
+static std::vector<ValueObjectSP>
+ExtractChildrenFromSwiftPointerValueObject(ValueObjectSP valobj_sp,
+                                           SwiftUnsafeType &unsafe_ptr) {
+  if (!valobj_sp)
+    return {};
+
+  lldb::ProcessSP process_sp(valobj_sp->GetProcessSP());
+
+  if (!process_sp)
+    return {};
+
+  const addr_t start_addr = unsafe_ptr.GetStartAddress();
+  const size_t num_children = unsafe_ptr.GetCount();
+  const CompilerType element_type = unsafe_ptr.GetElementType();
+
+  auto stride = element_type.GetByteStride(process_sp.get());
+  if (!stride)
+    return {};
+
+  auto element_stride = *stride;
+  size_t buffer_size = num_children * element_stride;
+  if (buffer_size > 512 * 1024 * 1024) {
+    LLDB_LOG(GetLog(LLDBLog::DataFormatters),
+             "Suspiciously large object: num_children={0}, stride={1}",
+             num_children, element_stride);
+    return {};
+  }
+
+  std::unique_ptr<lldb_private::WritableDataBuffer> buffer_up(
+      new DataBufferHeap(buffer_size, 0));
+  Status error;
+  size_t read_bytes = process_sp->ReadMemory(start_addr, buffer_up->GetBytes(),
```

Comment:
Why do we need to actually read the memory? Why can't we change the Value in the ValueObject to a LoadAddress?

Reply:
Not sure, I just extracted this function from `UnsafeTypeSyntheticFrontEnd::Update`

Reply:
Okay, I guess we can also address that later.